### PR TITLE
Enable changing webapp/server port using only env variable

### LINKED ISSUE #4049: Add an environment variable to set the web app port
## Tell us about the problem you're trying to solve
I'm trying to set up a Portainer stack using the git repository without having to download and edit docker-compose.yaml file.

## Describe the solution you’d like
Have the ability to set the web app port to something else than 8000 via an environment variable.

## Additional context

Portainer provides a convenient GUI to manage Docker containers, images, networks, etc. Via its "stacks" you can load a series of containers directly from Github, which is what I'm trying to do with Airbyte:
image

https://documentation.portainer.io/v2.0/stacks/create/

### PR DESCRIPTION
**Pull request by marcosmarxm**
## What
Closes #4049

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Secrets are annotated with `airbyte_secret` in output spec
- [ ] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [ ] Code reviews completed
- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [ ] Documentation updated 
    - [ ] README
    - [ ] CHANGELOG.md
    - [ ] Reference docs in the `docs/integrations/` directory.
    - [ ] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [ ] Build is successful
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] No major blockers
- [ ] PR merged into master branch
- [ ] Follow up tickets have been created
- [ ] Associated tickets have been closed & stakeholders notified
</p>
</details>

<details><summary> <strong> Connector Generator checklist </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -17,7 +17,9 @@ TRACKING_STRATEGY=segment
...
 # Issue: https://github.com/airbytehq/airbyte/issues/577
 HACK_LOCAL_ROOT_PARENT=/tmp
-WEBAPP_URL=http://localhost:8000/
+WEBAPP_PORT=8000
+SERVER_PORT=8001
+WEBAPP_URL=http://localhost:${WEBAPP_PORT}/
 API_URL=/api/v1/
 TEMPORAL_HOST=airbyte-temporal:7233
-INTERNAL_API_HOST=airbyte-server:8001
+INTERNAL_API_HOST=airbyte-server:${SERVER_PORT}
```
Comment (michel-tricot) [inline .env:25]: does variable interpolation work in the .env file?

How does this change work with @jrhizor modification to server everything from a single port?

**Review by jrhizor — CHANGES_REQUESTED** [review]
There are a few problems with this. The variable interpolation doesn't work in .env like Michel said, so you'd have to re-specify the port. A comment above the group of variables that need to be changed would be helpful.

Also `ServerApp` needs to be updated since it uses a constant. We'd need to add it to `EnvConfigs`.

The `.env.dev` file should also reflect these changes so it'd be easier to test locally.

We also may want to merge this close to a release so there isn't an interim period where there looks like a configurable port in .env but it isn't actually usable.

Before merge we should check that the dev version works end to end for a sync as expected with a new port.

**Comment by marcosmarxm** [comment]
> There are a few problems with this. The variable interpolation doesn't work in .env like Michel said, so you'd have to re-specify the port. A comment above the group of variables that need to be changed would be helpful.

Can I break into `WEBAPP_PORT` and  `WEBAPP_HOST` and unify them to `WEBAPP_URL` in the code? Something like:
```
  @Override
  public String getWebappUrl() {
    return getEnsureEnv(WEBAPP_HOST) + "/" + getEnsureEnv(WEBAPP_PORT);;
  }
```
 
> The `.env.dev` file should also reflect these changes so it'd be easier to test locally.
After changes, I'll test using `dev` building.

Thanks for the feedback.

**Comment by jrhizor** [comment]
I think `WEBAPP_URL` is only used for alerting, so it's probably best to allow that to be set directly. In a more productionized environment it's likely that they don't want specify a port at al land it's just something like `https://airbyte.companysite.com/`. The same concern exists for the api server. They may want to use an absolute path for the server too. For example, they could map the api server to a different route at some point like `https://api.companysite.com/etl-apis/airbyte/`. There's an issue for someone that wants to do something like that.

**Review by jrhizor — COMMENTED** [review]
I think I was leading you a bit astray here. Thanks @michel-tricot for pointing that out. 

For https://github.com/airbytehq/airbyte/issues/4049 specifically:
- you need the `WEBAPP_PORT` you have (but it might be better to call it `WEBAPP_EXTERNAL_PORT` or `WEBAPP_EXPOSED_PORT`. 
- A comment should make it clear that `WEBAPP_URL` also needs to be updated if `WEBAPP_PORT` is changed
- the docker compose file needs to be updated with `- ${WEBAPP_PORT}:80`

It's easiest to ignore the server ports and leave it as is. Since we are encouraging people to use just the webapp even for the API now that it forwards the API, we can treat the internal API port as statically set to 8001. What you added to map the exposed server port actually doesn't require internal code changes. I think it will cause confusion and steer people in the wrong direction, when the right direction is to customize the exposed webapp port.

**Pull request closed without merging**